MiniDumper/MiniDumper.cs

```diff
@@ -178,6 +180,28 @@ public void PrintDebugString(OUTPUT_DEBUG_STRING_INFO outputDbgStrInfo)
             }
         }
 
+        public void MemoryCommitThreshold(int commitThreshold)
+        {
+            System.Timers.Timer aTimer = new System.Timers.Timer();
+            aTimer.Interval = 1000;
+            aTimer.Elapsed += (sender, e) => OnTimedEvent(sender, e, commitThreshold); ;
+            aTimer.AutoReset = true;
+            aTimer.Enabled = true;
+        }
+
+        private void OnTimedEvent(Object source, ElapsedEventArgs e, int commitThreshold)
+        {
+           var process = Process.GetProcessById(pid);
+            double privateMemory = (process.PrivateMemorySize64 / 1024) / 1024;
+            if (privateMemory >= commitThreshold)
+            {
+                var timer = source as System.Timers.Timer;
+                timer.Stop();
+                timer.Dispose();
```

Comment:
We should not dispose this timer as we will move it to the class field. And here we should also take into consideration the number of dumps the user wants (**-n** parameter).